File: backend/src/main/java/viagens_e_turismo/models/HotelPacote.java
package viagens_e_turismo.models;

import java.io.Serializable;
import java.time.LocalDate;

import com.fasterxml.jackson.annotation.JsonBackReference;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import lombok.Data;

@Data
@Entity
public class HotelPacote implements Serializable{
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;
    
    @ManyToOne
    @JoinColumn(name="pacote_id")
    @JsonBackReference
    private Pacote pacote;

    @ManyToOne
    @JoinColumn(name="hotel_id")
    private Hotel hotel;

    private String tipoDiaria;
    private int qtdDiarias;
    private LocalDate dataEntrada;
}

File: frontend/src/app/components/pacotes/pacote/hotelPacote.ts
import { Time } from "@angular/common"
import { Endereco } from "../../endereco/endereco"
import { Pacote } from "./pacote"
import { Hotel } from "../../hotel/hotel"

export interface HotelPacote {
  id?: number
  pacote: Pacote
  hotel: Hotel
  tipoDiaria: string
  qtdDiarias: number
  dataEntrada: Date
}
